test/jasmine/tests/transform_filter_test.js

```diff
@@ -320,6 +321,22 @@ describe('filter transforms calc:', function() {
         expect(out[0].y).toEqual([1]);
     });
 
+    it('should preserve gaps in data when `preservegaps` is turned on', function() {
+        var out = _transform([Lib.extendDeep({}, base, {
+            transforms: [{
+                type: 'filter',
+                preservegaps: true,
+                operation: '>',
+                value: 0,
+                target: 'x'
+            }]
+        })]);
+
+        expect(out[0].x).toEqual([undefined, undefined, undefined, undefined, 1, 2, 3]);
+        expect(out[0].y).toEqual([undefined, undefined, undefined, undefined, 2, 3, 1]);
+        expect(out[0].marker.color).toEqual([undefined, undefined, undefined, undefined, 0.2, 0.3, 0.4]);
+    });
+
```

Comment:
Are there any interesting cases to test with two filters applied in sequence having different `preservegaps` values?